pkg/issuer/acme/http/pod_test.go

```diff
@@ -252,3 +253,117 @@ func TestGetPodsForCertificate(t *testing.T) {
 		})
 	}
 }
+
+func TestMergePodObjectMetaWithPodTemplate(t *testing.T) {
+	const createdPodKey = "createdPod"
+	tests := map[string]solverFixture{
+		"should use labels and owner references from template": {
```

Comment:
This has raised a new thought.. do we really want to allow setting owner references too? I think our own controller should still always set the owner/controller of the created resource to the 'parent' Challenge resource. Perhaps this is yet another field to disallow 🙄.

Further, setting owner references actually has some privilege escalation issues. By being able to set an owner reference on an object, you are blocking deletion of whatever object you are 'owned by' (and so, if the resource you've added the ownerRef to also has a finalizer, it is possible to lock up a cluster altogether).

Red Hat/OpenShift actually has some additional RBAC rules that are required in order to set owner references on *any* resource type. I think we should be careful to expose this, as well as `finalizers`, as it could allow a user to create resources with fields set using cert-manager's own service account/rbac permissions.